File: src/main/java/com/dot/examinator/domain/UserExam.java
package com.dot.examinator.domain;

import javax.persistence.Column;
import javax.persistence.EmbeddedId;
import javax.persistence.Entity;
import javax.persistence.Table;
import java.time.LocalDateTime;

@Entity
@Table(name = "user_exam")
public class UserExam {

    @EmbeddedId
    private UserExamId userExamId;

    @Column(name = "exam_date")
    private LocalDateTime examDate;
    @Column(name = "marks")
    private String marks;

    public UserExamId getUserExamId() {
        return userExamId;
    }

    public void setUserExamId(UserExamId userExamId) {
        this.userExamId = userExamId;
    }

    public LocalDateTime getExamDate() {
        return examDate;
    }

    public void setExamDate(LocalDateTime examDate) {
        this.examDate = examDate;
    }

    public String getMarks() {
        return marks;
    }

    public void setMarks(String marks) {
        this.marks = marks;
    }
}

File: exam-ui/src/app/domain/UserExam.ts
import { UserExamId } from '../domain/UserExamId';
export class UserExam {

    userId: number;
    examId: number;
    attempt: number;
    examDate: string;
    marks: string;
    userExamId: UserExamId;

    constructor(userId: number, examId: number, attempt: number, examDate: string, marks: string, userExamId: UserExamId) {
        this.userId = userId;
        this.examId = examId;
        this.attempt = attempt;
        this.examDate = examDate;
        this.marks = marks;
        this.userExamId = userExamId;
      }
}
